<?xml version="1.0"?>

<launch>
  <arg name="node_name" default="quadrotor0_xtion_voxel_grid" />
  <arg name="depth_pcl_unfiltered" default="quadrotor0/depth_pcl_unfiltered" />
  <arg name="voxel_grid_output" value="$(arg node_name)/output" />
  <arg name="depth_pcl_filtered" default="voxel_grid/depth_pcl"            />

 <node pkg="nodelet" type="nodelet" name="$(arg node_name)" args="load pcl/VoxelGrid pcl_manager" output="screen">
  <remap from="~input" to="$(arg depth_pcl_unfiltered)" />
  <remap from="$(arg voxel_grid_output)" to="$(arg depth_pcl_filtered)" />

  <rosparam>
      filter_field_name: z
      filter_limit_min: -0.5
      filter_limit_max: 10
      filter_limit_negative: False
      leaf_size: 0.1
  </rosparam>
 </node>



</launch>
